app/models/magic.py

```diff
@@ -0,0 +1,31 @@
+import time
+from dataclasses import dataclass
+
+
+@dataclass
+class Magic:
+    name: str
+    base_power: int
+
+    def get_power(self) -> int:
+        """
+        Обчислює магічну силу для Ланселота в залежності від поточного часу.
+
+        Часові діапазони:
+        - З 12:00 по 17:59: +15 до сили.
+        - З 18:00 по 23:59: 0 до сили.
+        - З 00:00 по 05:59: -10 до сили.
+        - З 06:00 по 11:59: 0 до сили.
+
+        Повертає значення магічної сили.
+        """
+        current_hour = time.localtime().tm_hour
```

Comment:
You are directly using the `time` module to get the current time. While this is not inherently wrong, it's generally a good practice to use a more abstract way of getting the current time, such as defining a function or method for this. This makes your code easier to test, as you can mock this function or method in your tests to return a specific time.